tick/optim/model/src/model_generalized_linear.cpp

```diff
@@ -95,6 +95,36 @@ double ModelGeneralizedLinear::loss(const ArrayDouble &coeffs) {
       / n_samples;
 }
 
+
+void ModelGeneralizedLinear::sdca_primal_dual_relation(const double l_l2sq,
```

Comment:
The abstract method might be in `ModelWithDuality` but this default primal dual relation works only with GLM imo. Hawkes relation wont fit in this framework for example as it needs to take psi into account (just like Poisson with identity link)